src/components/UserSelector.tsx

```diff
@@ -1,19 +1,64 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { getUsersAsync } from '../features/userSlice';
+import { setAuthor } from '../features/authorSlice';
+import { User } from '../types/User';
+import { setSelectedPost } from '../features/selectedPostSlice';
 
 export const UserSelector: React.FC = () => {
+  const [expanded, setExpanded] = useState(false);
+  const dispatch = useAppDispatch();
+
+  const selectedUser = useAppSelector(store => store.author.author);
+
+  useEffect(() => {
+    dispatch(getUsersAsync());
+  }, []);
+
+  useEffect(() => {
+    if (!expanded) {
+      return;
+    }
+
+    const handleDocumentClick = () => {
+      setExpanded(false);
+    };
+
+    document.addEventListener('click', handleDocumentClick);
+
+    // eslint-disable-next-line consistent-return
+    return () => {
+      document.removeEventListener('click', handleDocumentClick);
+    };
+  }, [expanded]);
+
+  const handleChange = useCallback((author: User | null | undefined) => {
+    dispatch(setAuthor(author));
+    dispatch(setSelectedPost(null));
+  }, []);
+
+  const users = useAppSelector(store => store.users.users);
+
   return (
     <div
       data-cy="UserSelector"
-      className="dropdown is-active"
+      className={classNames('dropdown', { 'is-active': expanded })}
     >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={(e) => {
+            e.stopPropagation();
+            setExpanded(current => !current);
+          }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here